olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -113,11 +114,26 @@ DefaultCache::StorageOpenResult DefaultCacheImpl::Open() {
   return SetupStorage();
 }
 
+DefaultCacheImpl::~DefaultCacheImpl() { Close(); }
+
 void DefaultCacheImpl::Close() {
   std::lock_guard<std::mutex> lock(cache_lock_);
   if (!is_open_) {
     return;
   }
+  if (mutable_cache_) {
+    auto batch = std::make_unique<leveldb::WriteBatch>();
+
+    auto value = protected_keys_.WriteListOfProtectedKeys();
+    if (value->size() > 0) {
+      leveldb::Slice slice(reinterpret_cast<const char*>(value->data()),
+                           value->size());
+      batch->Put(kProtectedKeys, slice);
+      auto result = mutable_cache_->ApplyBatch(std::move(batch));
+      OLP_SDK_LOG_DEBUG_F(kLogTag, "Store list of protected keys result=%s",
+                          result.IsSuccessful() ? "true" : "false");
+    }
+  }
```

Comment:
I would also consider writing the protected list to cache on Protect if it changed.

Reply:
marking list dirty on Protect and Release. Writing on put, or on close